File: Estrutura/Pilha.py
class Stack:
    def __init__(self):
        self.stack = []
        self.top = -1

    def push(self, item):
        self.top += 1
        self.stack.insert(self.top,item)
    def stack_empty(self):
        if self.top == -1:
            return True;
        else:
            return False;
    def pop(self):
        if self.stack_empty():
            return None
        item = self.stack[self.top]
        self.top -= 1
        self.stack.pop()
        return item

    def peek(self):
        if self.top == -1:
            return None
        return self.stack[self.top]


if __name__ == '__main__':
    stack = Stack()
    stack.push(1)
    stack.push(2)
    stack.push(3)
    print(stack.pop())  # Output: 3
    print(stack.pop())  # Output: 2
    print(stack.pop())  # Output: 1
    stack.push(10)
    print(stack.pop())  # Output: 10

File: Estrutura/Pilha.c
#include <stdio.h>
#include <stdlib.h>
#include<stdbool.h>
#define MAX_SIZE 10

int stack[MAX_SIZE];
int top = -1;

bool stack_empty()
{
    if(top == 0)
        return true;
    else return false;
}
void push(int x) {
    if (top == MAX_SIZE - 1) {
        printf("Pilha cheia - Stack overflow!\n");
        return;
    }
    top++;
    stack[top] = x;
}

int pop() {
    if (stack_empty()) {
        printf("Pilha vazia - Stack underflow!\n");
        return -1;
    }
    else
    {
        top--;
        return stack[top+1];
    }

}

int main() {
    push(1);
    push(2);
    push(3);
    push(4);
    push(5);

    int value = pop();
    printf("valor retirado da pilha: %d\n", value);

    return 0;
}

